src/components/activity/ActivityHistory.tsx

```diff
@@ -81,13 +89,15 @@ export const ActivityHistory = ({route, navigation}: any) => {
       const toTimeString = getFormattedTime(toDate);
 
       historyItems.push(
-        <Surface key={'hi_' + fromDate.toString()} style={{ flexDirection: 'row', borderRadius: 5, elevation: elevation.medium, marginHorizontal: 10 , marginVertical: 5 }}>
-          <List.Item style={{flex: 1, flexGrow: 1 }} title={fromTimeString + ' - ' + toTimeString} description={activity.text} right={() =>
-            <View style={{flexDirection: 'row', alignItems: 'center'}}>
-              <Caption>{activity.importance + ' | ' + activity.enjoyment}</Caption>
-              <List.Icon icon={activity.icon}/>
-            </View>
-          }/>
+        <Surface key={'hi_' + fromDate.toString()} style={{ borderRadius: 5, elevation: elevation.medium, marginHorizontal: 10 , marginVertical: 5 }}>
+          <TouchableRipple onPress={() => onModifyDate(day, activity)} style={{ flexDirection: 'row' }}>
+            <List.Item style={{flex: 1, flexGrow: 1 }} title={fromTimeString + ' - ' + toTimeString} description={activity.text} right={() =>
+              <View style={{flexDirection: 'row', alignItems: 'center'}}>
+                <Caption>{activity.importance + ' | ' + activity.enjoyment}</Caption>
+                <List.Icon icon={activity.icon}/>
+              </View>
+            }/>
+          </TouchableRipple>
```

Comment:
Instead of pushing a whole element, only the title and the index of an entry should be pushed. With that information the flatlist can render the components and some of the logic in ActivityRegistrator can be simplified.